packages/matplotlib/src/browser_backend.py

```diff
@@ -1,8 +1,67 @@
 import math
+from typing import Any, Callable
 
 from matplotlib.backend_bases import FigureCanvasBase, NavigationToolbar2, TimerBase
 
-from js import document, window
+from js import clearInterval, clearTimeout, document, setInterval, setTimeout
+from pyodide import JsProxy, create_once_callable, create_proxy
+
+try:
+    from js import devicePixelRatio as DEVICE_PIXEL_RATIO
+except ImportError:
+    DEVICE_PIXEL_RATIO = 1
+
+
+EVENT_LISTENERS: dict[tuple[JsProxy, str, JsProxy], JsProxy] = {}
+
+
+def add_event_listener(elt: JsProxy, event: str, listener: Callable[[Any]]):
+    proxy = create_proxy(listener)
+    EVENT_LISTENERS[(elt, event, listener)] = proxy
+    elt.addEventListener(event, proxy)
+
+
+def remove_event_listener(elt: JsProxy, event: str, listener: Callable[[Any]]):
+    proxy = EVENT_LISTENERS[(elt, event, listener)]
+    elt.removeEventListener(event, proxy)
+    proxy.destroy()
+
+
+class Destroyable:
+    def destroy(self):
+        pass
+
+
+TIMEOUTS: dict[int, Destroyable] = {}
+
+
+def set_timeout(callback: Callable[[]], timeout: int) -> int:
+    id = None
+
+    def wrapper():
+        nonlocal id
+        callback()
+        del TIMEOUTS[id]
+
+    id = setTimeout(create_once_callable(wrapper), timeout)
+    return id
+
+
+DUMMY_DESTROYABLE = Destroyable()
```

Comment:
Could you add a comment explaining the logic with this, please?

Reply:
It's just an object with a no-op destroy method so I can do `TIMEOUTS.pop(id, DUMMY_DESTROYABLE).destroy()` and either it gets a real object and calls the real `destroy` method or it gets the fake which does nothing. This is to handle the case where `clear_timeout` is called after the timeout executes.